week_3/scripts/ballotDeploy.ts

```diff
@@ -0,0 +1,45 @@
+import { ethers } from 'hardhat'
+import { Ballot__factory } from '../typechain-types';
+import * as dotenv from 'dotenv'
+
+dotenv.config()
+
+async function main() {
+    
+    //In the command line, we provide the token address with which we will be able to vote
+    //Then the target block number from which we want to count the voting power
+    //Then the proposals to be voted 
+    const args = process.argv;
+    const tokenAddress = args[2];
+    const numberOfBlocksBeforeVotingPowerCount = args[3];
+    const proposals = args.slice(4);
+
+    const provider = new ethers.providers.AlchemyProvider("goerli", process.env.ALCHEMY_API_KEY);
+    const privateKey = process.env.PRIVATE_KEY;
+    if(!privateKey || privateKey.length <= 0) {
+        throw new Error("Private key missing")
+    }
+    const wallet = new ethers.Wallet(privateKey);
+    const signer = wallet.connect(provider);
+
+    console.log("Deploying Tokenized Ballot contract");
+    console.log("Proposals: ");
+    proposals.forEach((element, index) => {
+      console.log(`Proposal N. ${index + 1}: ${element}`);
+    });
+  
+    const currentBlockNumber = await provider.getBlockNumber();
+    const ballotFactory = new Ballot__factory(signer);
+    const ballotContract = await ballotFactory.deploy(
+        proposals.map(proposal => ethers.utils.formatBytes32String(proposal)), 
+        tokenAddress,
+        currentBlockNumber + Number(numberOfBlocksBeforeVotingPowerCount)
```

Comment:
Could you please explain to me please the block number reflected here? If I understand right, `targetBlockNumber` in the Ballot contract is the block used to calculate voting power for each voter. Why do we have the current block number there as well?

Reply:
We have the current block number here because we are deploying on a testnet (goerli) then we add a number of blocks to it to make it work in the futur.

Reply:
is this because we are only minting tokens for voters after we deploy the ballot?

Reply:
We are giving everyone times to delegate their token since the token contract is also new